identity-core/src/common/timestamp.rs

```diff
@@ -21,44 +19,59 @@ use crate::error::Result;
 #[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
 #[repr(transparent)]
 #[serde(try_from = "String", into = "String")]
-pub struct Timestamp(DateTime<Utc>);
+pub struct Timestamp(OffsetDateTime);
 
 impl Timestamp {
-  /// Parses a `Timestamp` from the provided input string.
+  /// Parses a `Timestamp` from the provided input string, normalized to UTC+00:00 with fractional
+  /// seconds truncated.
+  ///
+  /// See the [`datetime` DID-core specification](https://www.w3.org/TR/did-core/#production).
   pub fn parse(input: &str) -> Result<Self> {
-    let datetime: DateTime<Utc> = DateTime::parse_from_rfc3339(input)?.into();
-    let datetime: DateTime<Utc> = Self::truncate(datetime);
-
-    Ok(Self(datetime))
+    let offset_date_time = OffsetDateTime::parse(input, &Rfc3339)
+      .map_err(time::Error::from)?
+      .to_offset(UtcOffset::UTC);
+    Ok(Timestamp(truncate_fractional_seconds(offset_date_time)))
   }
 
-  /// Creates a new `Timestamp` with the current date and time.
+  /// Creates a new `Timestamp` with the current date and time, normalized to UTC+00:00 with
+  /// fractional seconds truncated.
+  ///
+  /// See the [`datetime` DID-core specification](https://www.w3.org/TR/did-core/#production).
   pub fn now_utc() -> Self {
-    Self(Self::truncate(Utc::now()))
-  }
-
-  /// Returns the `Timestamp` as a Unix timestamp.
-  #[allow(clippy::wrong_self_convention)]
-  pub fn to_unix(&self) -> i64 {
-    self.0.timestamp()
-  }
-
-  /// Creates a new `Timestamp` from the given Unix timestamp.
-  pub fn from_unix(seconds: i64) -> Self {
-    Self(DateTime::from_utc(NaiveDateTime::from_timestamp(seconds, 0), Utc))
+    Self(truncate_fractional_seconds(OffsetDateTime::now_utc()))
   }
 
   /// Returns the `Timestamp` as an RFC 3339 `String`.
   ///
   /// See: https://tools.ietf.org/html/rfc3339
-  #[allow(clippy::wrong_self_convention)]
   pub fn to_rfc3339(&self) -> String {
-    self.0.to_rfc3339_opts(SecondsFormat::Secs, true)
+    // expect is OK, because all constructors provide well defined RFC 3339 compatible OffSetDateTime
+    // if this is to be made fallible instead then we need to change some things in our interface such as the
+    // From<Timestamp> for String implementation.
+    self
+      .0
+      .format(&Rfc3339)
+      .expect("unexpected Timestamp incompatibility with RFC 3339")
   }
 
-  fn truncate(value: DateTime<Utc>) -> DateTime<Utc> {
-    // Safe to unwrap because 0 is a valid nanosecond
-    value.with_nanosecond(0).unwrap()
+  /// Returns the `Timestamp` as a Unix timestamp.
+  pub fn to_unix(&self) -> i64 {
+    self.0.unix_timestamp()
+  }
+
+  /// Creates a new `Timestamp` from the given Unix timestamp.
+  ///
+  /// # Errors
+  /// Fails if `seconds` is outside of the interval [-62167219200,253402300799].
```

Comment:
Optional, trying to give more context.

```suggestion
  /// Creates a new `Timestamp` from the given Unix timestamp.
  ///  
  /// The timestamp must be in the valid range for [RFC 3339](https://tools.ietf.org/html/rfc3339).
  ///
  /// # Errors
  /// [`Error::InvalidTimestamp`] if `seconds` is outside of the interval [-62167219200,253402300799].
```